packages/arb-avm-cpp/cmachine/arbcore.go

```diff
@@ -273,17 +273,18 @@ func (ac *ArbCore) LogsCursorRequest(cursorIndex *big.Int, count *big.Int) error
 	return nil
 }
 
-func (ac *ArbCore) LogsCursorGetLogs(cursorIndex *big.Int) ([]value.Value, error) {
+func (ac *ArbCore) LogsCursorGetLogs(cursorIndex *big.Int) (*big.Int, []value.Value, error) {
 	cursorIndexData := math.U256Bytes(cursorIndex)
-	result := C.arbCoreLogsCursorGetLogs(ac.c, unsafeDataPointer(cursorIndexData))
+	var firstIndex big.Int
+	result := C.arbCoreLogsCursorGetLogs(ac.c, unsafeDataPointer(cursorIndexData), unsafeDataPointer(firstIndex.Bytes()))
```

Comment:
I don't think that writing to `firstIndex` is safe since it's a complex type. Instead we'd need a [32]byte to write into

Reply:
Looks like `LogsCursorGetDeletedLogs` has the same issue

Reply:
took care of by returning value in struct